apps/02RockPaperScissors.js

```diff
@@ -5,17 +5,48 @@ var prompt = require('prompt');
 prompt.start();
 
 
-function rockPaperScissors(hand1, hand2) {
+function rockPaperScissors (hand1, hand2) {
 
-    // Write code here
+  //check for a tie
+    if (hand1 === hand2) {
+      return "It's a tie!";
+    }
+
+    if (hand1 === 'paper') {
+      if (hand2 === 'rock') {
+        return 'Hand one wins!';
+      } else if (hand2 === 'scissors') {
+        return 'Hand two wins!';
+      }
+    }
+    if (hand1 === 'rock') {
+      if (hand2 === 'paper') {
+        return 'Hand two wins!';
+      } else if (hand2 === 'scissors') {
+        return 'Hand one wins!';
+      }
+    }
+    if (hand1 === 'scissors') {
+      if (hand2 === 'paper') {
+        return 'Hand one wins!';
+      } else if (hand2 === 'rock') {
+        return 'Hand two wins!';
+      }
+    }
+  }
+
+  function isValidRPSInput(input) {
+    return !(input !== 'rock' && input !== 'paper' && input !== 'scissors');
+  }
 
-}
 
 function getPrompt() {
     prompt.get(['hand1', 'hand2'], function (error, result) {
-
+      if(isValidRPSInput(result['hand1']) && isValidRPSInput(result['hand2'])) {
```

Comment:
I tried running this on my machine but it kept looping getPrompt once I submitted an entry..

![image](https://cloud.githubusercontent.com/assets/15767173/17671073/dde50816-62c9-11e6-9c41-85d36279b3cd.png)